src/components/Metronome/Metronome.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Flex } from '../Box';
+import { Text } from '../Text';
+import { useMetronome } from './useMetronome';
+import { Flasher } from './Flasher';
+
+function Metronome({ startMs, metronomeNumber, beatsPerMinutes, sx, ...props }) {
+  /*
+   * Any way I can extract the index number to the array out of these definitions and pass it in
+   * when referenced?
+   */
+  const [whole] = useMetronome(beatsPerMinutes[metronomeNumber] / 4, startMs);
+  const [half] = useMetronome(beatsPerMinutes[metronomeNumber] / 2, startMs);
+  const [quarter] = useMetronome(beatsPerMinutes[metronomeNumber], startMs);
+  const [eighth] = useMetronome(beatsPerMinutes[metronomeNumber] * 2, startMs);
+  const [sixteenth] = useMetronome(beatsPerMinutes[metronomeNumber] * 4, startMs);
+  const [thirtysecond] = useMetronome(beatsPerMinutes[metronomeNumber] * 8, startMs);
+
+  /*
+   * The following almost-duplicated definitions go away if I can pass in the array index
+   */
+  const [whole1] = useMetronome(beatsPerMinutes[metronomeNumber + 1] / 4, startMs);
+  const [half1] = useMetronome(beatsPerMinutes[metronomeNumber + 1] / 2, startMs);
+  const [quarter1] = useMetronome(beatsPerMinutes[metronomeNumber + 1], startMs);
+  const [eighth1] = useMetronome(beatsPerMinutes[metronomeNumber + 1] * 2, startMs);
+  const [sixteenth1] = useMetronome(beatsPerMinutes[metronomeNumber + 1] * 4, startMs);
+  const [thirtysecond1] = useMetronome(beatsPerMinutes[metronomeNumber + 1] * 8, startMs);
+
+  const [whole2] = useMetronome(beatsPerMinutes[metronomeNumber + 2] / 4, startMs);
+  const [half2] = useMetronome(beatsPerMinutes[metronomeNumber + 2] / 2, startMs);
+  const [quarter2] = useMetronome(beatsPerMinutes[metronomeNumber + 2], startMs);
+  const [eighth2] = useMetronome(beatsPerMinutes[metronomeNumber + 2] * 2, startMs);
+  const [sixteenth2] = useMetronome(beatsPerMinutes[metronomeNumber + 2] * 4, startMs);
+  const [thirtysecond2] = useMetronome(beatsPerMinutes[metronomeNumber + 2] * 8, startMs);
+
+  /*
+   * I want to be able to get the constants defined above as elements of the array as
+   *  defined in the property types
+   */
```

Comment:
@digable1 

I think we can do this. Probably the thing we can do is have a pair programming session where I talk you through how to do what you want.